CorsixTH/Lua/map.lua

```diff
@@ -402,9 +404,33 @@ function Map:loadMapConfig(filename, config, custom)
         end
       end
     end
+
+    -- If the level file overlay field gives a difficulty and number (from the campaign),
+    --   try to load that on top of the config loaded so far.
+    if config.overlay then
+      local difficulty, level_number = config.overlay.difficulty, config.overlay.level_number
+      local errors, _
+      config.overlay = nil -- Prevent recursive loop
+
+      if difficulty and level_number then
+        -- First overlay the difficulty
+        errors, config = self:loadMapConfig(difficulty .. "00.SAM", config)
+        if errors then return "Overlay difficulty must be easy, full or hard. Current value is " .. difficulty end
+        if type(level_number) == "number" and level_number < 10 then
+          level_number = "0" .. level_number
+        end
+        -- Overlay with the specific configuration for this level
+        errors, config = self:loadMapConfig(difficulty .. level_number .. ".SAM", config)
+        if errors then return "Overlay level number must be 1-12. Current value is " .. level_number end
+        -- Finally overlay additional CorsixTH config per level
+        local level_path = self.app:getFullPath({"Levels", "original" .. level_number .. ".level"})
+        _, config = self:loadMapConfig(level_path, config, true)
+      else return "No difficulty and level number given in overlay field of custom level"
+      end
+    end
     return nil, config
   else
-    return "Error: Could not find the configuration file, only 'Base Config' will be loaded for this level.", config
+    return "Error: Could not find the configuration file.", config
```

Comment:
Are we no longer trying to load base_config as a final fallback Toby?

Reply:
Yes, where we attempt to load Corsixth changes to the campaign is the only time we want to ignore failure. Anywhere else, it’s a fatal error that the user (a level writer) should see.

Reply:
Add a comment to explain that?

Exceptions to a general rule raise questions about correctness, unless there is clear evidence it's intentional.

Reply:
There is one in the loadMapConfig doc, now adding comments to the calls where failure is ignored.

This now warns the user if a campaign level file is missing, which is only needed as we don't check all the files at start. It is meant for end users, and could be a language string. It's being strict, where we may have previously papered over with the base config.